doc/source/WorkingWithThePythonScriptingAPI/ThePythonEditor/index.md

```diff
@@ -0,0 +1,78 @@
+```eval_rst
+.. role:: raw-html(raw)
+    :format: html
+```
+
+# The Python Editor #
+
+The _Python Editor_ is a type of editor in the main Gaffer application that can execute arbitrary Python code with a built-in interpreter. It enables you to modify the node graph, try commands and syntax, and inspect data. In the default layout, it resides in the bottom-right panel, in a tab next to the _Hierarchy View_.
+The _Python Editor_ is split into two areas. The bottom-half is the input field. The top-half is the output. The input field of the _Python Editor_ functions like a basic plain text editor. You can type, select, cut, copy, paste, and execute code.
+
+![A Python Editor with its input and the output field](images/interfacePythonEditorOutput.png "A Python Editor with its input and the output field")
+
+The output field behaves like a typical console output. It records the output stream, including return values and errors. Errors appear in red.
+
+![Errors in a Python Editor's output field](images/interfacePythonEditorError.png "Errors in a Python Editor's output field")
+
+> Tip :
```

Comment:
Would this tip be better placed later in the article? We haven't demonstrated any commands yet and it talks about adding nodes to the graph. It feels like it should come after 'Executing code' at least?

Reply:
At this stage, we're just introducing the editor's elements. "Seeing what an error looks like" isn't a task. Admittedly this might be better served in a general UI article (of which we don't have any yet), but I'd like to leave it.

Reply:
Sorry this is GitHubs's irritating context quoting. My comment was really on the tip about having to import modules to be able to add nodes to a grab. Error docs = good!

Reply:
Ah, that makes more sense. Good point.

Reply:
I ended up moving this part into the main text, and explaining how each _Python Editor_ is a separate instance.